server/message_hooks.go

```diff
@@ -541,6 +542,8 @@ func (p *Plugin) SendChat(srcUser string, usersIDs []string, post *model.Post, c
 	}
 
 	p.GetMetrics().ObserveMessage(metrics.ActionCreated, metrics.ActionSourceMattermost, true)
+	p.GetMetrics().ObserveMessageDelay(metrics.ActionCreated, metrics.ActionSourceMattermost, true, newMessage.CreateAt.Sub(time.UnixMilli(post.CreateAt)))
```

Comment:
Here, we're measuring the time between when we Teams received the message (`newMessage.CreateAt`, returned via the API) and the time we created the post in Mattermost.